pkgs/os-specific/linux/kernel/manual-config.nix

```diff
@@ -111,7 +114,7 @@ let
         if [ -f scripts/gcc-plugins/gen-random-seed.sh ]; then
           substituteInPlace scripts/gcc-plugins/gen-random-seed.sh \
             --replace NIXOS_RANDSTRUCT_SEED \
-            $(echo ${src} ${configfile} | sha256sum | cut -d ' ' -f 1 | tr -d '\n')
+            $(echo ${randstructSeed} ${src} ${configfile} | sha256sum | cut -d ' ' -f 1 | tr -d '\n')
```

Comment:
you may remove space in `${randstructSeed} ${src}` and this won't trigger kernel (and pkgs) rebuild. Maybe like
```
${lib.concatMapStringsSep " " builtins.toString [randstructSeed src configfile]}
``` 
or maybe just `${configfile}` -> `${randstructSeed + configfile}`